From 571542b44f072e97b1f336f30b6dc02e52f865bd Mon Sep 17 00:00:00 2001
Message-Id: <571542b44f072e97b1f336f30b6dc02e52f865bd.1691350701.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1691350701.git.sharlatanus@gmail.com>
References: <cover.1691350701.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 6 Aug 2023 14:30:23 +0100
Subject: [PATCH 1/9] gnu: Add go-github-com-bitly-go-hostpool.

* gnu/packages/golang.scm (go-github-com-bitly-go-hostpool): New variable.
---
 gnu/packages/golang.scm | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index ea6aadbe80..0a747f2eee 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -25,7 +25,7 @@
 ;;; Copyright © 2020 Martin Becze <mjbecze@riseup.net>
 ;;; Copyright © 2021, 2022 Ricardo Wurmus <rekado@elephly.net>
 ;;; Copyright © 2021 Guillaume Le Vaillant <glv@posteo.net>
-;;; Copyright © 2021 Sharlatan Hellseher <sharlatanus@mgail.com>
+;;; Copyright © 2021, 2023 Sharlatan Hellseher <sharlatanus@mgail.com>
 ;;; Copyright © 2021 Sarah Morgensen <iskarian@mgsn.dev>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
 ;;; Copyright © 2021 jgart <jgart@dismail.de>
@@ -2112,6 +2112,31 @@ (define-public go-github-com-dhowett-go-plist
 types.")
       (license license:giftware))))
 
+(define-public go-github-com-bitly-go-hostpool
+  (package
+    (name "go-github-com-bitly-go-hostpool")
+    (version "0.1.0")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/bitly/go-hostpool")
+                    (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+              (sha256
+               (base32
+                "1iibj7dwymczw7cknrh6glc6sdpp4yap2plnyr8qphynwrzlz73w"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/bitly/go-hostpool"))
+    (native-inputs (list go-github-com-stretchr-testify))
+    (home-page "https://github.com/bitly/go-hostpool")
+    (synopsis "Pool among multiple hosts from Golang")
+    (description
+     "This package provides a Go package to intelligently and flexibly pool among
+multiple hosts from your Go application.  Host selection can operate in round
+robin or epsilon greedy mode, and unresponsive hosts are avoided.")
+    (license license:expat)))
+
 (define-public go-github-com-blanu-dust
   (package
     (name "go-github-com-blanu-dust")
-- 
2.40.1

